docs/testplan/platform/pmon-test-plan.md

```diff
@@ -1,113 +1,227 @@
-#### Test objective
-Verify the expected data for thermalctld daemon (if not skip_thermalctld)
-#### Test steps
-* Verify the policy file "/usr/share/sonic/platform/thermal_policy.json" exists
-* Verify the 'FAN_INFO|{}' and 'TEMPERATURE_INFO|{}' available
-   * Verify the data to check with the output of the platform api 
-
-### Test case # 8 – Data for xcvrd daemon
-#### Test objective
-Verify the expected data for xcvrd daemon (if not skip_xcvrd)
-#### Test steps
-* Verify the 'TRANSCEIVER_INFO', 'TRANSCEIVER_DOM_SENSOR' and 'TRANSCEIVER_STATUS' state_db table has data
-* Verify the information data for each 'TRANSCEIVER_INFO' is expected to compare "/usr/share/sonic/platform/platform.json"
-   * Verify the data to check with the output of the platform api 
-
-## TODO
-
-## Open questions
+### Test steps
+1. Parse /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2,
+   /usr/share/sonic/platform/pmon_daemon_control.json, and /etc/supervisor/conf.d/supervisord.conf
+2. Find the list of daemons running in PMON docker for the specific platform of DUT
+   a. Check all available daemons in SONiC image
+      - Check the output of "sudo docker exec pmon bash -c '[ -f /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2 ] \
+      && cat /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2 | grep program'"
+      - Add the daemons from the output as "all_daemons" list
+      ```
+   admin@sonic:~$ sudo docker exec pmon bash -c '[ -f /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2 ]  && cat /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2 | grep program'
+   [program:rsyslogd]
+   [program:chassisd]
+   [program:lm-sensors]
+   [program:fancontrol]
+   [program:ledd]
+   [program:pcied]
+   [program:psud]
+   [program:syseepromd]
+   [program:thermalctld]
+   [program:xcvrd]
+      ```
+   b. Find any daemon that doesn't support by the platform 
+      - Check the output of "sudo docker exec pmon bash -c '[ -f /usr/share/sonic/platform/pmon_daemon_control.json ] \
+      && cat /usr/share/sonic/platform/pmon_daemon_control.json'"
+      - Add the skip_*** as "skipped_daemons" list
+      ```
+   admin@sonic:~$ sudo docker exec pmon bash -c '[ -f /usr/share/sonic/platform/pmon_daemon_control.json ] \
+   && cat /usr/share/sonic/platform/pmon_daemon_control.json '
+   {
+       "skip_ledd": true,
+       "skip_thermalctld": true
+   }
+      ```
+   c. Check whether the DUT (platform) has configuration files which the specific daemon can start with.
+      - Check if the platform configuration files exists
+         <202012 master has three configuration files available to start the daemons as below.>
+         * chassisd : "/usr/share/sonic/platform/chassisdb.conf"
+         * lm-sensors : "/usr/share/sonic/platform/sensors.conf"
+         * fancontrol : "/usr/share/sonic/platform/fancontrol"
+      - Add the daemon as "skipped_daemons" list if the platform configuration file doesn't exist for a daemon to start.
```

Comment:
Should we consider these daemons skipped, or should this instead be a failure case, because the daemon couldn't start because the appropriate config file was not present? I'm not sure we want to consider a daemon "skipped" unless it is explicitly skipped in pmon_daemon_control.json.

Reply:
But these three files are checked inside the pmon supervisord configuration and if they are not there then pmon doesn't even start those daemon/services.
Can we just give some warnings that the platform doesn't have those files instead of marking as a failure case? And I don't think the single ASIC platform has chassisdb.conf. 
```
{% if not skip_chassisd and IS_MODULAR_CHASSIS == 1 %}
[program:chassisd]
command=/usr/local/bin/chassisd
...
{% if not skip_sensors and HAVE_SENSORS_CONF == 1 %}
[program:lm-sensors]
command=/usr/bin/lm-sensors.sh
...
{% if not skip_fancontrol and HAVE_FANCONTROL_CONF == 1 %}
[program:fancontrol]
command=/usr/sbin/fancontrol
```